src/core/catalog.js

```diff
@@ -1682,6 +1682,18 @@ class ExtendedCatalog extends Catalog {
     if (isDict(el) && el.has('Type') && el.get('Type').name === 'MCR') {
       return {mcid: el.get('MCID'), pageIndex: page};
     }
+
+    if (isDict(el) && !el.has('Pg') && !el.has('K')) {
```

Comment:
It's more or less obvious that `el` at this point doesn't have the 'K' field. I would suggest to check only if `el` is a Dict and has 'S' field. For us it's a sign that it is marked as a structure element.